### ISSUE TITLE
Critical error when using jetpack/13.0-a.1

### ISSUE THREAD

**Issue by helenaartmann** [Automattic/jetpack#34627]
### Impacted plugin

Jetpack

### Quick summary

Jetpack was automatically updated and when the user deactivated Classic Editor plugin, the critical error appeared on the site.

We found a temporary workaround by changing the PHP version to 7.4. 

### Steps to reproduce

1. Jetpack was automatically updated to 13.0-a.1
2. When the user deactivated the Classic Editor plugin, a critical error appeared on the site.
3. To see the error, we had to go to the pages and click on editing a page
4. <img width="822" alt="Screenshot 2023-12-13 at 9 46 46 AM" src="https://github.com/Automattic/jetpack/assets/72891246/63078484-dfca-42cd-865c-6a279ce8fe7a">

### A clear and concise description of what you expected to happen.

Be able to edit the page without the error appearing. 

### What actually happened

We don't have access to edit the page. 

### Impact

One

### Available workarounds?

Yes, easy to implement

### Logs or notes
CLI error: 
```
[13-Dec-2023 17:48:19 UTC] PHP Fatal error:  Uncaught TypeError: Cannot access offset of type string on string in /wordpress/plugins/jetpack/13.0-a.1/jetpack_vendor/automattic/jetpack-publicize/src/jetpack-social-settings/class-settings.php:155
Stack trace:
#0 /wordpress/plugins/jetpack/13.0-a.1/class.jetpack-gutenberg.php(743): Automattic\Jetpack\Publicize\Jetpack_Social_Settings\Settings->get_settings(true)
#1 /wordpress/core/6.4.2/wp-includes/class-wp-hook.php(324): Jetpack_Gutenberg::enqueue_block_editor_assets('')
#2 /wordpress/core/6.4.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#3 /wordpress/core/6.4.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#4 /wordpress/core/6.4.2/wp-includes/script-loader.php(2446): do_action('enqueue_block_a...')
#5 /wordpress/core/6.4.2/wp-includes/class-wp-hook.php(324): wp_common_block_scripts_and_styles('post.php')
#6 /wordpress/core/6.4.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#7 /wordpress/core/6.4.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#8 /wordpress/core/6.4.2/wp-admin/admin-header.php(118): do_action('admin_enqueue_s...', 'post.php')
#9 /wordpress/core/6.4.2/wp-admin/edit-form-blocks.php(309): require_once('/wordpress/core...')
#10 /wordpress/core/6.4.2/wp-admin/post.php(187): require('/wordpress/core...')
#11 {main}
  thrown in /wordpress/plugins/jetpack/13.0-a.1/jetpack_vendor/automattic/jetpack-publicize/src/jetpack-social-settings/class-settings.php on line 155
```

**Comment by helenaartmann** [comment]
7447661-zd-a8c - follow up

**Comment by jeherve** [comment]
cc @gmjuhasz who worked on #34113. Could you take a look? This was only a warning in PHP 7.4 but was bumped to an error on PHP 8.0.

Thanks!

**Comment by Greatdane** [comment]
[7448953-zd-a8c](7448953-zd-a8c)
[7449292-zd-a8c](7449292-zd-a8c)
[7445308-zd-a8c](7445308-zd-a8c)

Thom also shared more info on the issue here;
p9F6qB-dTZ-p2#comment-56925

**Comment by cuemarie** [comment]
I just had a customer with this issue here: 7453477-zen ; it was isolated to their third-party theme `ad-mania`, and occurring when they tried accessing the Customizer. Downgrading to PHP 7.4 resolved.

@gmjuhasz  is there anything we should inform customers about that they would need to bring to their theme authors?  Or should #34636 resolve this for any site? (also asked this here: p9F6qB-dTZ-p2#comment-56956)

**Comment by pablinos** [comment]
This got closed by merging the PR, but it won't be fixed until it's released on Atomic. 

The fix is to delete the offending option:

`wp option delete jetpack_social_autoconvert_images`

And then not save settings on the General page `/wp-admin/options-general.php` until the fix is released. Downgrading PHP is working because it's more lax with things like treating strings as arrays!

> Or should https://github.com/Automattic/jetpack/pull/34636 resolve this for any site?

Yes, once the fix is deployed it will be fixed for all sites.

**Comment by paxblueribbon** [comment]
Issue came up in zdc-7454795, resolved by rolling back php version

**Comment by Greatdane** [comment]
7455700-zd-a8c

**Comment by zdenys** [comment]
7458133-zd-a8c

**Comment by sdixon194** [comment]
Just noting this fix was about to be deployed before other issues surfaced, so we needed another fix on top of it. I'll cut another version today but it might not be deployed until next week at this rate.

**Comment by dhackneya8c** [comment]
another report in 7459451-zd-a8c

**Comment by mdbitz** [comment]
latest Jetpack has been released to Atomic

**Comment by anomiex** [comment]
Closing this per the above, the fix should be on Atomic with 13.0-a.5.
